src/libs/ajax.js

```diff
@@ -1153,20 +1153,21 @@ const Buckets = signal => ({
     return res.json()
   },
 
-  listAll: async (googleProject, bucket, prefix = null, pageToken = null) => {
+  listAll: async (googleProject, bucket, prefix = null, pageToken = null, delimiter = null) => {
```

Comment:
What is the delimiter for?

Reply:
I know we have very little method documentation, but I personally think it would be nice to document the input arguments and output of the methods (like I did for `getSpendReport`). Given this is not generally our style, it's up to you.

Reply:
Because directories are "fake" in GCS, supplying a delimiter allows us to list objects in a way that's easier to preserve the directory structure that users are expecting. From the [GCS documentation](https://cloud.google.com/storage/docs/json_api/v1/objects/list):

Returns results in a directory-like mode, with / being a common value for the delimiter.

`items[]` contains object metadata for objects whose names do not contain `delimiter`, or whose names only have instances of `delimiter` in their `prefix`.

`prefixes[]` contains truncated object names for objects whose names contain `delimiter` after any `prefix`. Object names are truncated beyond the first applicable instance of the delimiter. If multiple objects have the same truncated name, duplicates are omitted.


I'll add some documentation to this code, and a link to the GCS documentation.

Reply:
With the large number of arguments for this function, it might be nice to move the optional ones into an "options object".

```js
listAll: async (googleProject, bucket, { prefix = null, pageToken = null, delimiter = null } = {}) => {
...
}
```

So that something like `listAll('project', 'bucket', null, null, '/')` becomes `listAll('project', 'bucket', { delimiter: '/' })`.

Reply:
I like it. Done in https://github.com/DataBiosphere/terra-ui/pull/2921/commits/15d46e356aefc877cf2cd0b1cf12e7e1617d8085